src/main/java/org/folio/inventory/resources/Items.java

```diff
@@ -93,10 +95,15 @@ public void register(Router router) {
     router.put(RELATIVE_ITEMS_PATH_ID).handler(this::update);
     router.delete(RELATIVE_ITEMS_PATH_ID).handler(this::deleteById);
 
-    router.post(RELATIVE_ITEMS_PATH_ID + "/mark-in-process")
-      .handler(handle(this::markAsInProcess));
-    router.post(RELATIVE_ITEMS_PATH_ID + "/mark-in-process-non-requestable")
-      .handler(handle(this::markAsInProcessNonRequestable));
+    Arrays.stream(ItemStatusName.values())
+      .map(x->ItemStatusURL.getUrlForItemStatusName(x))
+      .filter(x->x.isPresent())
+      .forEach(x->router.post(RELATIVE_ITEMS_PATH_ID+x.get()).handler(handle(this::markItemAsTargetStatus)));
```

Comment:
I think it would be easier to understand this code if it was extracted to a method with an intent revealing name. Maybe something like:

```suggestion
      .forEach(itemStatusName -> registerHandlerForChangingItemStatus(itemStatusName, router);
```